pairing_api/src/pairing.rs

```diff
@@ -0,0 +1,472 @@
+        &self,
+        topic: &str,
+        params: ResponseParamsSuccess,
+        message_id: MessageId,
+        client: &Client,
+    ) -> Result<(), PairingClientError> {
+        let irn_metadata = params.irn_metadata();
+        let response = Response::Success(SuccessfulResponse {
+            id: message_id,
+            jsonrpc: JSON_RPC_VERSION_STR.into(),
+            result: serde_json::to_value(params)
+                .map_err(|err| PairingClientError::EncodeError(err.to_string()))?,
+        });
+
+        // Publish the encrypted message
+        self.publish_payload(topic, irn_metadata, Payload::Response(response), client)
+            .await?;
+
+        println!("\nOutbound request sent!");
+
+        Ok(())
+    }
+
+    /// Private function to publish a payload.
+    async fn publish_payload(
+        &self,
+        topic: &str,
+        irn_metadata: IrnMetadata,
+        payload: Payload,
+        client: &Client,
+    ) -> Result<(), PairingClientError> {
+        // try to extend session before updating local store.
+        let sym_key = {
+            let pairings = self.pairings.lock().await;
+            let pairing = pairings
+                .get(topic)
+                .ok_or_else(|| PairingClientError::PairingNotFound)?;
+            hex::decode(pairing.sym_key.clone()).map_err(|err| {
+                PairingClientError::EncodeError(format!("Failed to decode sym_key: {:?}", err))
+            })?
+        };
+
+        let payload = serde_json::to_string(&payload)
+            .map_err(|err| PairingClientError::EncodeError(err.to_string()))?;
+        let message = encrypt_and_encode(EnvelopeType::Type0, payload, &sym_key)
+            .map_err(|e| anyhow::anyhow!(e))
+            .map_err(|err| PairingClientError::EncodeError(err.to_string()))?;
+
+        // Publish the encrypted message
+        {
+            client
+                .publish(
+                    topic.into(),
+                    message,
+                    None,
+                    irn_metadata.tag,
+                    Duration::from_secs(irn_metadata.ttl),
+                    irn_metadata.prompt,
+                )
+                .await
+                .map_err(PairingClientError::PingError)?;
+        };
+
+        Ok(())
+    }
+
+    /// Private function to generate a WalletConnect URI.
+    fn generate_uri(pairing: &PairingInfo, sym_key: &str) -> String {
+        let methods_str = pairing
+            .methods
+            .0
+            .iter()
+            .map(|method_group| format!("[{}]", method_group.join(",")))
+            .collect::<Vec<_>>()
+            .join(",");
+        let expiry_timestamp_str = pairing.expiry.to_string();
+
+        format!(
+            "wc:{}@{}?symKey={}&methods={}&relay-protocol={}&expiryTimestamp={}",
+            pairing.topic,
+            VERSION,
+            sym_key,
+            methods_str,
+            pairing.relay.protocol,
+            expiry_timestamp_str
```

Comment:
you can pass `pairing.expiry` directly, format! macro automatically handles types that implement the Display trait

Reply:
resolved